src/Abstractions/NexusMods.Abstractions.Downloads/ADownloadActivity.cs

```diff
@@ -0,0 +1,101 @@
+using JetBrains.Annotations;
+using NexusMods.Abstractions.Activities;
+using NexusMods.MnemonicDB.Abstractions;
+using NexusMods.Paths;
+using ReactiveUI;
+using ReactiveUI.Fody.Helpers;
+
+namespace NexusMods.Abstractions.Downloads;
+
+/// <summary>
+/// Implementation of <see cref="IDownloadActivity"/> for ease of use.
+/// </summary>
+[PublicAPI]
+public abstract class ADownloadActivity : ReactiveObject, IDownloadActivity
+{
+    private readonly IConnection _connection;
+
+    /// <inheritdoc/>
+    public PersistedDownloadStateId PersistedStateId { get; }
+
+    /// <inheritdoc/>
+    public IDownloader Downloader { get; }
+
+    /// <inheritdoc/>
+    public string Title { get; }
+
+    /// <inheritdoc/>
+    public AbsolutePath DownloadPath { get; }
+
+    /// <summary>
+    /// Constructor.
+    /// </summary>
+    protected ADownloadActivity(
+        PersistedDownloadState.ReadOnly persistedState,
+        IDownloader downloader,
+        AbsolutePath downloadPath)
+    {
+        _connection = persistedState.Db.Connection;
+
+        PersistedStateId = persistedState.PersistedDownloadStateId;
+        _status = persistedState.Status;
+
+        Downloader = downloader;
+        Title = persistedState.Title;
+        Downloader = downloader;
+    }
+
+    private PersistedDownloadStatus _status;
+
+    /// <inheritdoc/>
+    public PersistedDownloadStatus Status
+    {
+        get => GetStatus();
+        set => SetStatus(value);
+    }
+
+    /// <inheritdoc/>
+    [Reactive] public Size BytesTotal { get; set; }
+
+    /// <inheritdoc/>
+    [Reactive] public Size BytesDownloaded { get; set; }
+
+    /// <inheritdoc/>
+    [Reactive] public Size BytesRemaining { get; set; }
+
+    /// <inheritdoc/>
+    [Reactive] public Percent Progress { get; set; }
+
+    /// <inheritdoc/>
+    [Reactive] public Bandwidth Bandwidth { get; set; }
+
+    /// <summary>
+    /// Gets the current status.
+    /// </summary>
+    protected virtual PersistedDownloadStatus GetStatus() => _status;
```

Comment:
Does this even return the latest value?

Reply:
Yes.

Reply:
Unless you mean the latest value from the DB. That's not really a concern, anything that mucks with the state outside this activity would be considered undefined behavior.

Reply:
So this would be the sole and only place with where Status should ever be edited? Specifically through `DownloadActivity.Status` public property?

Reply:
Yup

Reply:
I guess this isn't the intuitive place I would use to update the status. 
Since this interface is public facing (e.g. we will end up using these in the UI) I don't think it should be possible to set the Status through the public interface at least.  

I would have this be implemented by observing the DB and not even allow setting it.


I would assume IDownloaders would be the ones in charge of updating this, inside the Pause(), Cancel(), Start() calls

Reply:
Specifically, it would only be updated by the Downloader.

Reply:
I can remove the setters from `IDownloadActivity`, you'd still have the setters in `ADownloadActivity` and the downloaders would have to cast to that.

Reply:
Well the functionality doesn't strictly need to be on the activity, you just need the entity ID and anyone (the Downloader) can do it.

And the Getter would need to return the latest DB value, not the internal field value.